File: gitdot-api-ts/src/resource/task.ts
import { z } from "zod";

export const TaskResource = z.object({
  id: z.uuid(),
  repository_id: z.uuid(),
  build_id: z.uuid(),
  s2_uri: z.string(),
  name: z.string(),
  command: z.string(),
  status: z.string(),
  waits_for: z.array(z.uuid()),
  created_at: z.iso.datetime(),
  updated_at: z.iso.datetime(),
});
export type TaskResource = z.infer<typeof TaskResource>;

export const PollTaskResource = z.object({
  id: z.uuid(),
  repository_id: z.uuid(),
  s2_uri: z.string(),
  name: z.string(),
  command: z.string(),
  status: z.string(),
});
export type PollTaskResource = z.infer<typeof PollTaskResource>;

File: gitdot-api/src/resource/task.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use uuid::Uuid;

#[derive(ApiResource, Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
pub struct TaskLogResource {
    pub seq_num: u64,
    pub timestamp: u64,
    pub body: String,
    pub stream: Option<String>,
    pub finished: Option<String>,
}

#[derive(ApiResource, Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
pub struct TaskResource {
    pub id: Uuid,
    pub repository_id: Uuid,
    pub build_id: Uuid,
    pub s2_uri: String,
    pub name: String,
    pub command: String,
    pub status: String,
    pub waits_for: Vec<Uuid>,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

#[derive(ApiResource, Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
pub struct PollTaskResource {
    pub id: Uuid,
    pub token: String,

    pub owner_name: String,
    pub repository_name: String,
    pub s2_uri: String,

    pub name: String,
    pub command: String,
    pub status: String,
}
